examples/cosmos/rollkit_test.go

```diff
@@ -0,0 +1,221 @@
+				Denom:            "stake",
+				CoinType:         "118",
+				SigningAlgorithm: "",
+				GasPrices:        "0stake",
+				GasAdjustment:    2.0,
+				TrustingPeriod:   "112h",
+				NoHostMount:      false,
+				SkipGenTx:        false,
+				PreGenesis:       nil,
+				ModifyGenesis: func(config ibc.ChainConfig, bytes []byte) ([]byte, error) {
+					valKeyFileBz, _, err := rollkitChain.Validators[0].Exec(ctx, []string{"cat", "/var/cosmos-chain/gm/config/priv_validator_key.json"}, []string{})
+					if err != nil {
+						return nil, err
+					}
+					var pvKey privval.FilePVKey
+					if err = cmtjson.Unmarshal(valKeyFileBz, &pvKey); err != nil {
+						return nil, err
+					}
+
+					newGenesis := append(sdk50Genesis, cosmos.GenesisKV{
+						Key: "consensus.validators",
+						Value: []map[string]interface{}{
+							{
+								"address": pvKey.Address.String(),
+								"pub_key": map[string]interface{}{
+									"type":  "tendermint/PubKeyEd25519",
+									"value": base64.StdEncoding.EncodeToString(pvKey.PubKey.Bytes()),
+								},
+								"power": "1000",              // This is important
+								"name":  "Rollkit Sequencer", // This is important
+							},
+						},
+					})
+
+					daHostName := rollkitChain.Sidecars[0].HostName()
+
+					if _, _, err := rollkitChain.Sidecars[0].Exec(ctx, []string{"mkdir", "/home/celestia/bridge"}, []string{}); err != nil {
+						return nil, err
+					}
+
+					daAuthTokenBz, _, err := rollkitChain.Sidecars[0].Exec(ctx, []string{"celestia", "bridge", "auth", "admin", "--node.store", "/home/celestia/bridge"}, []string{})
+					if err != nil {
+						return nil, err
+					}
+					daAuthToken := strings.TrimSuffix(string(daAuthTokenBz), "\n")
+
+					if _, _, err := rollkitChain.Validators[0].Exec(ctx, []string{"bash", "-c", fmt.Sprintf(`echo "[rollkit]
+da_address = \"http://%s:%s\"
+da_auth_token = \"%s\"" >> /var/cosmos-chain/gm/config/config.toml`, daHostName, "26658", daAuthToken)}, []string{}); err != nil {
+						return nil, err
+					}
+
+					return cosmos.ModifyGenesis(newGenesis)(config, bytes)
+				},
+				ModifyGenesisAmounts: func(_ int) (sdk.Coin, sdk.Coin) {
+					return sdk.NewInt64Coin("stake", 10_000_000_000_000), sdk.NewInt64Coin("stake", 1_000_000_000)
+				},
+				ConfigFileOverrides: nil,
+				EncodingConfig:      nil,
+				UsingChainIDFlagCLI: false,
+				SidecarConfigs: []ibc.SidecarConfig{
+					{
+						ProcessName: "local-celestia-devnet",
+						Image: ibc.DockerImage{
+							Repository: "ghcr.io/rollkit/local-celestia-devnet",
+							Version:    "latest",
+							UidGid:     "1025:1025",
+						},
+						HomeDir: "/home/celestia",
+						Ports: []string{
+							"26650/tcp",
+							"26657/tcp",
+							"26658/tcp",
+							"26659/tcp",
+							"9090/tcp",
+						},
+						StartCmd: []string{"/bin/bash", "/opt/entrypoint.sh"},
+						Env:      nil, // Here we could set CELESTIA_NAMESPACE if needed
+						PreStart: true,
+						StartCheck: func(index int) error {
```

Comment:
Maybe to be more inline with https://github.com/strangelove-ventures/interchaintest/pull/1123, we rename this to PostStart() on the sidecar process. Just so it more closely matches that PRs API conventions and also is more versatile vs just a standard check.

Does not seem the `index` here is used? Could put the CosmosChain here or something

Reply:
That naming makes sense.

Yeah, the idea was originally to send in the SidecarProcess but that gets to a cyclical dependency so ended up with an index and then in the end didn't use it. 

In the code below I just do Sidecars[0], but I guess the idea was to access the right one. But I suppose I would always know the order they are added (being the same as the SidecarConfigs slice order). So could send in the chain like you said instead.